README.md

```diff
@@ -1,155 +1,236 @@
 
-    // trigger to notify external listeners that new data is available
-    trigger();
+**Important:** an action has no business logic. It is simply a notification system for stores.
+
+
+## **2. Stores respond to dispatched Actions**
+
+A `Store` responds to actions, applies the appropriate business logic, updates the model, and broadcasts a change event. The base Store class provided by flutter_flux should be extended to fit the needs of your app and its data. App state may be spread across many independent stores depending on the complexity of the app and your desired app architecture.
+
+By convention, a `Store`'s internal data cannot be mutated directly. Instead, `Store` data is mutated internally in response to `Action` dispatches. `Store`s should otherwise be considered read-only, publicly exposing relevant data ONLY via getter methods. This limited data access ensures that the integrity of the uni-directional data flow is maintained.
+
+A `Store` is **not** the model - a `Store` contains the model, which is usually either a List or a `Map` of a single type. In this case, `ToDoStore` is a `List` of `Todo`.
+
+```dart
+class ToDo {
+  ToDo(this.toDoText);
+  String toDoText;
+  Bool isComplete = false;
+}
+
+class ToDoStore extends Store {
+  final _todos = new List<ToDo>();
+  List<ToDo> get todos => new List.unmodifiable(_todos);
+}
+```
+
+To handle an `Action`, a `Store` subscribes to the `Action` and defines a handler function, like this:
+
+```dart
+class ToDoStore extends Store {
+  ToDoStore() {
+    triggerOnAction(submitToDoAction, (InputValue value) {
+      _todos.add(new ToDo(value.text));
+    });
   }
+
+  final _todos = new List<ToDo>();
+  List<ToDo> get todos => new List.unmodifiable(_todos);
 }
 ```
 
-**BONUS:** `Stores` can be initialized with a stream transformer to modify the standard behavior of the `trigger` stream.
-This can be useful for throttling UI rendering in response to high frequency `Store` mutations.
+With this change, every time a widget dispatches `submitToDoAction`, the text of the new item will be added to the _todos list.
+
+You instantiate a store using a StoreToken, like this:
 
 ```dart
-import 'package:rate_limit/rate_limit.dart';
-import 'package:flutter_flux/flutter_flux.dart';
+final StoreToken todoStoreToken =
+    new StoreToken(new ToDoStore());
+```
+
+Hiding the `ToDoStore` inside of a `StoreToken` enforces the limited visibility of a `Store`. The `Store` will be explicitly passed to your `build()` function and can referenced by its token, as shown in the next section.
 
-class ThrottledStore extends Store {
-  ...
+We could add the trivial business logic to ignore empty ToDo items:
 
-  ThrottledStore(this._actions) : super.withTransformer(new Throttler(const Duration(milliseconds: 30))) {
-    ...
+```dart
+class ToDoStore extends Store {
+  ToDoStore() {
+    triggerOnAction(submitToDoAction, (InputValue value) {
+      if (value.text != null && value.text.trim().isNotEmpty) {
```

Comment:
maybe use null-aware operators here?

I think it's value?.text.trim().isNotEmpty